pkg/engine/validate/utils.go

```diff
@@ -20,25 +22,36 @@ func hasNestedAnchors(pattern interface{}) bool {
 		}
 		return false
 	case []interface{}:
+		fmt.Println("[]interface{}......")
 		for _, value := range typed {
 			if hasNestedAnchors(value) {
 				return true
 			}
 		}
 		return false
 	default:
+		fmt.Println("default......")
 		return false
 	}
 }
 
 // getSortedNestedAnchorResource - sorts anchors key
 func getSortedNestedAnchorResource(resources map[string]interface{}) *list.List {
 	sortedResourceKeys := list.New()
+	fmt.Println("\n-----------getSortedNestedAnchorResource------------")
+	fmt.Println("resources: ", resources)
 	for k, v := range resources {
+		fmt.Println("k: ", k, "                 v:", v)
+		if commonAnchors.IsGlobalAnchor(k) {
+			sortedResourceKeys.PushFront(k)
+			continue
+		}
```

Comment:
I think we need to clarify:
1. whether the nested global anchor is allowed, take the following rule as an example, if the resource only uses `nginx` image and does not have any `env` variable defined, do we validate the `imagePullSecrets`?
```yaml
spec: 
  rules: 
  - name: test
    match: 
      resources: 
        kinds: 
        - Pod
    validate: 
      pattern: 
        spec: 
          containers: 
          - name: "*"
            <(image): nginx
            env: 
            - <(name): foo
              <(value): bar
          imagePullSecrets:
          - my-registry-secret
```
2. can we combine global anchor and other types of anchor together? If yes, we cannot just continue the key has the global anchor defined.

Reply:
I have added a few more test cases, including the above pattern. 
The above given pattern passes the test.

Reply:
@realshuting - What are your thoughts on this? should it always check the global anchor and then proceed for the other anchors?

Reply:
As discussed, I have remove the test cases where we are using two validation anchors.